### PR TITLE
Improve release workflow

### PR DESCRIPTION
**Pull request by coolreader18**
Basically now, if you want to create a release, push a commit to `main` that has `release X.Y.Z` somewhere in the subject line. Then, in the commit message body, put whatever changelog or text you want to go in the body of the github release.

See [https://github.com/coolreader18/Celeste64/releases/tag/0.5.0](an example release) generated by this workflow, and [the workflow run](https://github.com/coolreader18/Celeste64/actions/runs/8331955551) that did it.

### CODE REVIEW

File: .github/workflows/build-stable.yml [hunk 1]
```diff
@@ -94,33 +55,81 @@ jobs:
...
+    steps:
+      - name: Publish Github Release
+        uses: ncipollo/release-action@v1
+        with:
+          tag: ${{ needs.release_info.outputs.version }}
+          commit: "${{ github.sha }}"
+          allowUpdates: true
+          omitBodyDuringUpdate: true
+          omitNameDuringUpdate: true
+          draft: false
```
Comment (jasminegamedev) [inline .github/workflows/build-stable.yml:135]: I think we'll want to release it as a draft so we can review it and test it and make finishing touches to the release in github before actually publishing it. So we might not need this publishing step?

**Review by jasminegamedev — CHANGES_REQUESTED** [review]
I'm not the best at github actions yml, but I've skimmed over this, and I think it looks good and makes sense from what I understand. (assuming it works, but everything looked good to me from your test release)
I think the only change I want to request is to not publish the release automatically when we're done, and just leave it as a draft. I think we're going to want to manually publish it step, so we can review it and clean it up before publishing it.

**Comment by jasminegamedev** [comment]
One additional question I just thought of. Since we are changing it to create the release before the builds happen, does that mean the draft will still get created, even if the build fails?
I wonder if we might want to see if we can structure it in a way to avoid that. 🤔
Not a hard requirement, but just something we might want to consider.

File: .github/workflows/build-stable.yml (hunk 1 above)
Comment (coolreader18) [reply .github/workflows/build-stable.yml:135]: Ah ok - makes sense!

**Comment by jasminegamedev** [comment]
I think this looks good to me now. We can see how it goes on our next release, and see if we need any more changes at that point. Merging.

**Pull request merged**